libsyclinterface/tests/test_sycl_queue_submit_local_accessor_arg.cpp

```diff
@@ -0,0 +1,364 @@
+//===-- test_sycl_queue_submit.cpp - Test cases for kernel submission fns. ===//
+//
+//                      Data Parallel Control (dpctl)
+//
+// Copyright 2020-2024 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file has unit test cases for the various submit functions defined
+/// inside dpctl_sycl_queue_interface.cpp.
+//===----------------------------------------------------------------------===//
+
+#include "dpctl_sycl_context_interface.h"
+#include "dpctl_sycl_device_interface.h"
+#include "dpctl_sycl_device_selector_interface.h"
+#include "dpctl_sycl_event_interface.h"
+#include "dpctl_sycl_kernel_bundle_interface.h"
+#include "dpctl_sycl_kernel_interface.h"
+#include "dpctl_sycl_queue_interface.h"
+#include "dpctl_sycl_type_casters.hpp"
+#include "dpctl_sycl_usm_interface.h"
+#include <filesystem>
+#include <fstream>
+#include <gtest/gtest.h>
+#include <sycl/sycl.hpp>
+#include <utility>
+
+namespace
+{
+constexpr size_t SIZE = 100;
+
+using namespace dpctl::syclinterface;
+
+typedef struct MDLocalAccessorTy
+{
+    size_t ndim;
+    DPCTLKernelArgType dpctl_type_id;
+    size_t dim0;
+    size_t dim1;
+    size_t dim2;
+} MDLocalAccessor;
```

Comment:
The need to define this in test file is a hint that we should have this `typedef` in a header file of libsyclinterface itself.